### DISCUSSION TITLE
How to specify table column width sizes?

### DISCUSSION THREAD

**Discussion by ckamps** [gethinode/hinode#711] [Help] [answered]
How can the size of table columns be specified either proportionally as relative percentage or by explicit size?

I have a lot of tables that represent features of a home that is being marketed. I’d prefer that the first column or name of each feature not be rendered with text/word wrap when displays of sufficient sizes are used.

For example, is it feasible to include Bootstrap class references in table columns when using Hinode? 

https://getbootstrap.com/docs/4.1/utilities/sizing/

Here’s an example where the first column is word wrapped:

<img width="1102" alt="image" src="https://github.com/gethinode/hinode/assets/1332810/1d3adee1-9693-4612-972b-a1247bccc9b9">

**Comment by markdumay** [comment, marked as answer]
Unfortunately cell-level styling or row-level styling is not supported. Hinode basically processes the output of the markdown generated by Hugo, which only supports block-level styling (see the section about `attribute` in the [goldmark markdown renderer docs](https://gohugo.io/getting-started/configuration-markup/#goldmark)).

What I would do instead is define a custom style and assign that as a class attribute to the table. For example, the following table shortcode applies the `my-table` style:

```markdown
{{< table class="my-table" >}}
[...]
{{< /table >}}
```

You can define the `my-table` class in a `scss` file such as `assets/scss/theme/theme.scss`:

```css
.my-table {
// your style definitions
}
```

Alternatively, you could enter the HTML code directly in your Markdown content file. You would need to set `unsafe = true` in the Hugo configuration:

```toml
[markup]
  [markup.goldmark]
    [markup.goldmark.renderer]
      unsafe = true
```

One final remark, you referenced the Bootstrap v4 documentation. Although it is similar to Bootstrap v5, it is not fully compatible. Hinode supports v5 only.
